src/WiFi.cpp

```diff
@@ -148,13 +148,31 @@ static void resolve_cb(uint8_t * /* hostName */, uint32_t hostIp)
 }
 
 static void ping_cb(uint32 u32IPAddr, uint32 u32RTT, uint8 u8ErrorCode) {
-  if (u8ErrorCode == 0)
-    if (WiFi._resolve == u32IPAddr)
-      WiFi._resolve = WL_PING_SUCCESS;
-    else
-      WiFi._resolve = WL_PING_DEST_UNREACHABLE;
-  else
-    WiFi._resolve = u8ErrorCode;
+	switch(u8ErrorCode){
```

Comment:
I think this was clearer as an `if` statement.

Also, I think we should be using the `PING_ERR_SUCCESS` , `PING_ERR_DEST_UNREACH`, and `PING_ERR_TIMEOUT ` constants defined in [src/socket/include/m2m_socket_host_if.h](https://github.com/arduino-libraries/WiFi101/blob/78bb2158676aec7936ab899185ee97a52705f70a/src/socket/include/m2m_socket_host_if.h#L182-L184)